filters/HAGFilter.cpp

```diff
@@ -103,9 +137,89 @@ void HAGFilter::filter(PointView& view)
     for (PointId i = 0; i < ngView->size(); ++i)
     {
         PointRef point = ngView->point(i);
+        double x0 = point.getFieldAs<double>(Dimension::Id::X);
+        double y0 = point.getFieldAs<double>(Dimension::Id::Y);
         double z0 = point.getFieldAs<double>(Dimension::Id::Z);
-        auto ids = kdi.neighbors(point, 1);
-        double z1 = gView->getFieldAs<double>(Dimension::Id::Z, ids[0]);
+        auto ids = kdi.neighbors(point, m_count);
+        double z1 = std::numeric_limits<double>::quiet_NaN();
+        assert(ids.size() > 0);
+        if (ids.size() == 1) {
+            z1 = gView->getFieldAs<double>(Dimension::Id::Z, ids[0]);
+        } else if (m_delaunay_fans == false) { // Nearest-Neighbor-based interpolation
+            auto min_x = std::numeric_limits<double>::max();
+            auto max_x = std::numeric_limits<double>::min();
+            auto min_y = std::numeric_limits<double>::max();
+            auto max_y = std::numeric_limits<double>::min();
+            double weights = 0;
+            double z_accumulator = 0;
+            bool exact_match = false;
+            for (unsigned long j = 0; j < ids.size(); ++j) {
```

Comment:
Why not use `size_t` for this and the other iterator variables?

Reply:
Certainly, will change.

Reply:
Okay, have changed